calendar_dst_bug_fix/models/calendar_event.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+
+import pytz
+from datetime import datetime, timedelta
+from odoo import api, models, fields as api_fields
+
+
+class CalendarEvent(models.Model):
+    _inherit = 'calendar.event'
+
+    @api.multi
+    def read(self, fields=None, load='_classic_read'):
+        TIME_FIELDS_IN_EVENT = ['start', 'stop',
+                                'start_datetime', 'stop_datetime']
+        result = super(CalendarEvent, self).read(fields, load)
+        data = []
+        tz = self.env.context and self.env.context.get('tz')
+        if tz:
+            tz = pytz.timezone(tz)
+        for record in result:
+            id_pair = str(record['id']).split('-')
+            if (not record.get('allday')
+                    and isinstance(record['id'], (str))
+                    and tz and len(id_pair) > 1):
+                delta = self.get_time_dst_delta(int(id_pair[0]), tz, record)
+                if delta:
```

Comment:
```suggestion
                if not delta:
                    continue
```

To avoid many interations

Reply:
@luistorresm that would mean to repeat the `data.append(record)` line right before the continue which doesn't seem DRY

Reply:
This is to avoid the next iteration.

If you only need to update when delta is defined, then you could continue when is not present.

With this, line 27-31 will be moved 4 spaces before

Reply:
the `data.append(record)` should be executed even if there's no delta so calling continue doesn't let that part of the code execute the code would have to look like:
```
...
                if not delta:
                    data.append(record)
                    continue
                for field in TIME_FIELDS_IN_EVENT:
                    if field in record and record[field]:
                        record[field] = api_fields.Datetime.to_string(
                            api_fields.Datetime.from_string(
                                record[field]) + delta)
            data.append(record)
...
```

Reply:
👍